### PR TITLE
Document failed rows analysis

### PR DESCRIPTION
**Pull request by janet-can**
Updated failed rows samples details in multiple places to include information about failed rows analysis for Soda Cloud accounts connected to a Soda Agent. 

See:
- Analyze failed rows
- Failed rows checks > Define failed rows checks
- Missing metrics > Failed row samples
- Validity metrics > Failed row samples
- Numeric metrics > Failed row samples
- Reference checks > Failed row samples
- Optional check configurations > Collect failed rows samples

### CODE REVIEW

**Comment by bastienboutonnet** [comment]
screenshot

I'm not sure I undestand that section. It brings us to user-defined failed rows set up. Which seems a bit of a different thing. Probably, worth a reword.


<details><summary>Browser metadata</summary>





```
Path:      /soda-cloud/failed-rows.html
Browser:   Chrome 111.0.0.0 on Mac OS 10.15.7
Viewport:  1323 x 1569 @2x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-493--jovial-piroshki-65ff4d.netlify.app%2Fsoda-cloud%2Ffailed-rows.html&os=OS%20X&os_version=Catalina&browser=Chrome&browser_version=111.0&resolution=1323x1569&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-493--jovial-piroshki-65ff4d.netlify.app/soda-cloud/failed-rows.html) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=63f54715ae3b22000856d5cd&commentID=642425498f78cc005d3126cd&resolution=resolved)

**Comment by bastienboutonnet** [comment]
Thanks for getting this started Janet it looks really good.

- Compatibility: Nice digging, you got those correct
- I think we’re missing a section on the fact that the failed rows **analysis will not work** for user-defined checks. This means that failed row checks where a `fail condition:` or `fail query:` is used will not be analysed. I propose the following explanation (feel free to edit it of course)

**Failed Rows Analysis on builtins only**
The Failed Rows Analysis is a feature in Soda that allows users to quickly identify and analyze the rows that have failed a data quality check. However, this feature is not available for user-defined failed rows checks that use a **`fail condition`**or **`fail query`**.
    
The reason for this limitation is that Soda needs to ensure that the data produced by the queries used to detect failing rows is suitable for analysis. Since users can use complex SQL queries and aggregations in their user-defined checks, it is not possible for Soda to guarantee that the data produced will be appropriate for analysis. As a result, the Failed Rows tab will only display the failed rows sample for user-defined checks that use a **`fail condition`**or **`fail query`**.
    

- Another thing that’s important to explain is that we only analyse categorical columns. Feel free to rework this explanation somewhere in the docs:
    
**Failed Rows Analysis on Categoriacals only**
It is important to note that the failed rows analysis in Soda only considers categorical columns, rather than scalar columns with continuous values. The reason for this limitation is that the failed rows analysis is designed to help users identify groups of rows that are affected by failed data quality checks, and categorical columns provide more natural segmentation points than continuous scalar columns. Additionally, since scalar columns can have values that span a wide range, it is not useful for humans to segment data by these columns. To identify the categorical columns, Soda uses a combination of machine learning and heuristics. The resulting subset of columns is then used to segment the failed rows for analysis. This means that users will only be able to segment the failed rows by a subset of the columns in the dataset, as determined by Soda's column selection process.
    

- Finally, not sure what you think, I think we could explain a bit more why and when users should use the failed rows analysis or what it’s useful for. Maybe this is better in a to-come blogpost? In either case, I have something that I was planning to riff off off in a blog or that would be a good announcement for community or news letter (below). Do you think we want to pick out some use cases in our docs at all?
    
**Failed Rows Analysis Use-cases**
The Failed Rows Analysis feature in Soda can be very useful for users who want to gain insights into the root causes of failed data quality checks. By analyzing the failed rows, users can identify patterns and trends that may be causing the data quality issues, and take steps to address them. Specifically, the failed rows analysis can help users:
    
    1. Identify specific groups of rows that are failing data quality checks: By analyzing the failed rows, users can identify specific subsets of data that are failing the checks, which can help to pinpoint the root cause of the issue.
    2. Understand patterns and trends in the failed data: Failed rows may exhibit certain patterns or trends that are not apparent in the data as a whole. By analyzing these patterns, users can gain insights into the underlying causes of the data quality issues.
    3. Prioritize data quality issues for remediation: By focusing on the subsets of data that are failing the most checks, users can prioritize their efforts to improve data quality and allocate resources accordingly.
    4. Improve data quality over time: By continually analyzing the failed rows and taking steps to address the underlying issues, users can improve data quality over time and ensure that their data is more accurate, reliable, and actionable.

File: soda-cloud/failed-rows.md [hunk 1]
```diff
@@ -1,20 +1,80 @@
...
+
+#### Troubleshoot
+
+{% include troubleshoot-failed-rows.md %}
+
+### Compatibility
+
+The following data sources support the failed row analysis feature.
+
+{% include compatible-datasources.md %}
```
[the remark this answers is not in the document]
Comment (bastienboutonnet) [reply soda-cloud/failed-rows.md:71]: @baturayo do you think that it isn't clear enough:

Screenshot 2023-03-29 at 15 22 20

**Comment by janet-can** [comment]
> Finally, not sure what you think, I think we could explain a bit more why and when users should use the failed rows analysis or what it’s useful for. Maybe this is better in a to-come blogpost?

Love it, and absolutely a candidate for a soon-to-come blog post.

Meanwhile, I'm incorporating the rest of your notes/comments into the documentation. Merci!

**Pull request closed without merging**